mm2src/mm2_main/src/swap_versioning.rs

```diff
@@ -0,0 +1,26 @@
+//! Swap Versioning Module
+//!
+//! This module provides a dedicated type for handling swap versioning
+
+#[derive(Copy, Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct SwapVersion {
+    pub version: u32,
+}
+
+impl Default for SwapVersion {
+    fn default() -> Self {
+        Self {
+            version: legacy_swap_version(),
+        }
+    }
+}
+
+impl SwapVersion {
+    pub(crate) fn is_legacy(&self) -> bool { self.version == legacy_swap_version() }
```

Comment:
```suggestion
impl SwapVersion {
    pub(crate) const fn is_legacy(&self) -> bool { self.version == legacy_swap_version() }
```

Reply:
Hmm why we can use const here if actual `self.version` value will be known at runtime?
I mean I can compile it, I thought it wont work. Is it bcz version has a primitive type?